File: dotnet/DictaTest/PerformanceTest.cs
using EdgeJs;
using System;
using System.IO;
using System.Threading.Tasks;

namespace DictaDotNet
{
    public class PerformanceTest
    {
        private void TestConcat()
        {
            Console.WriteLine("***\nTestConcat");
            string text = File.ReadAllText(@"dicta\coretest\performance\concat.dicta");
            Dicta model = new Dicta();
            model.Parse(text);
            int appendCount = (int)1e4;
            model.Set("count", appendCount);
            int runCount = 10;
            DateTime start = DateTime.Now;
            for (int runIndex = 1; runIndex <= runCount; runIndex++)
            {
                model.Set("base", runIndex);
                model.Get("a");
            }
            TimeSpan elapsed = DateTime.Now - start;
            Console.WriteLine(string.Format("Appended {0} times in {1} milliseconds", appendCount,
                elapsed.TotalMilliseconds));
            Console.WriteLine();
        }

        public void All()
        {
            TestConcat();
        }
    }
}

File: java/DictaTest/test/net/geocentral/dicta/test/PerformanceTest.java
package net.geocentral.dicta.test;

import java.util.Date;

import net.geocentral.dicta.Dicta;

import org.junit.Test;

public class PerformanceTest {

    @Test
    public void testConcat() throws Exception {
        System.out.println("\n***\ntestConcat");
        Dicta model = TestUtils.read("coretest/performance/concat");
        int appendCount = (int)1e4;
        model.set("count", appendCount);
        int runCount = 10;
        Date start = new Date();
        for (int runIndex = 0; runIndex < runCount; runIndex++) {
            model.set("base", runIndex + 1);
            model.get("a");
        }
        long elapsed = new Date().getTime() - start.getTime();
        System.out.println(String.format("Appended %s times in %s milliseconds", appendCount, elapsed/runCount));
    }
}
